app/models/deal.rb

```diff
@@ -1,5 +1,7 @@
 class Deal < ApplicationRecord
   has_many :deal_images, dependent: :destroy
+  has_many :line_items, dependent: :restrict_with_exception
```

Comment:
why exception here?

Reply:
should be error here,  as we don't want to stop the flow of application